### PR TITLE
Remove substation and it's voltage levels

### PR DESCRIPTION
**Pull request by souissimai**

### CODE REVIEW

File: src/test/java/org/gridsuite/modification/server/modifications/SubstationDeletionTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                .equipmentType("SUBSTATION")
+                .equipmentId("s1")
+                .build();
+    }
+
+    @Override
+    protected ModificationInfos buildModificationUpdate() {
+        return EquipmentDeletionInfos.builder()
+                .equipmentType("LINE")
+                .equipmentId("v2")
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/modification/server/modifications/SubstationDeletionTest.java:40]: Replace by `l1`

**Review by SlimaneAmar — APPROVED** [review]

File: src/test/java/org/gridsuite/modification/server/modifications/SubstationDeletionTest.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import org.gridsuite.modification.server.dto.ModificationInfos;
+import org.gridsuite.modification.server.utils.NetworkCreation;
+
+import java.util.UUID;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+/**
+ * @author Etienne Homer <etienne.homer at rte-france.com>
```
Comment (etiennehomer) [inline src/test/java/org/gridsuite/modification/server/modifications/SubstationDeletionTest.java:20]: It's your work, not mine :)

File: src/test/java/org/gridsuite/modification/server/ModificationControllerTest.java [hunk 3]
```diff
@@ -875,24 +881,48 @@ public void testTombstonedEquipmentInfos() throws Exception {
...
+        );
+        expectedImpacts.addAll(createMultipleDeletionImpacts(
+            List.of(
+                Pair.of(IdentifiableType.SWITCH, "v1bl3"), Pair.of(IdentifiableType.SWITCH, "v1dl3"),
+                Pair.of(IdentifiableType.SWITCH, "v4bl1"), Pair.of(IdentifiableType.SWITCH, "v4dl1")
+            ),
+            Set.of("s1")
+        ));
+        expectedImpacts.addAll(createMultipleDeletionImpacts(
+            List.of(Pair.of(IdentifiableType.LINE, "line1"), Pair.of(IdentifiableType.LINE, "line3")), new HashSet<>(Set.of("s1", "s2"))
```
Comment (dbraquart) [inline src/test/java/org/gridsuite/modification/server/ModificationControllerTest.java:919]:
```suggestion
            List.of(Pair.of(IdentifiableType.LINE, "line1"), Pair.of(IdentifiableType.LINE, "line3")), Set.of("s1", "s2")
```

File: src/test/java/org/gridsuite/modification/server/modifications/SubstationDeletionTest.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+    @Override
+    protected void assertNetworkAfterCreation() {
+        assertNull(getNetwork().getSubstation("s1"));
+        assertNull(getNetwork().getVoltageLevel("v1"));
+        assertNull(getNetwork().getVoltageLevel("v2"));
+        assertNull(getNetwork().getLoad("v1load"));
+        assertNull(getNetwork().getLccConverterStation("v1lcc"));
+        assertNull(getNetwork().getSwitch("v1d1"));
+        assertNull(getNetwork().getLine("line2"));
```
Comment (dbraquart) [inline src/test/java/org/gridsuite/modification/server/modifications/SubstationDeletionTest.java:52]:
```suggestion
        assertNull(getNetwork().getLine("line1"));
        assertNull(getNetwork().getLine("line2"));
```

File: src/test/java/org/gridsuite/modification/server/modifications/SubstationDeletionTest.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        assertNull(getNetwork().getHvdcLine("hvdcLine"));
+    }
+
+    @Override
+    protected void assertNetworkAfterDeletion() {
+        assertNotNull(getNetwork().getSubstation("s1"));
```
Comment (dbraquart) [inline src/test/java/org/gridsuite/modification/server/modifications/SubstationDeletionTest.java:58]:
```suggestion
        assertNotNull(getNetwork().getSubstation("s1"));
        assertNotNull(getNetwork().getVoltageLevel("v1"));
        assertNotNull(getNetwork().getVoltageLevel("v2"));
        assertNotNull(getNetwork().getLoad("v1load"));
        assertNotNull(getNetwork().getLccConverterStation("v1lcc"));
        assertNotNull(getNetwork().getSwitch("v1d1"));
        assertNotNull(getNetwork().getLine("line1"));
        assertNotNull(getNetwork().getLine("line2"));
        assertNotNull(getNetwork().getHvdcLine("hvdcLine"));
```

**Review by dbraquart — CHANGES_REQUESTED** [review]
code: ok, minor changes requested
tests: ok

**Review by dbraquart — APPROVED** [review]
code: ok
tests: not done with last commit

**Pull request merged**